introspection-engine/connectors/sql-introspection-connector/src/introspection.rs

```diff
@@ -133,29 +135,57 @@ pub(crate) fn introspect(ctx: &mut Context) -> Result<(), SqlError> {
         .table_walkers()
         .filter(|table| is_prisma_1_point_1_or_2_join_table(*table) || is_prisma_1_point_0_join_table(*table))
     {
-        calculate_fields_for_prisma_join_table(table, &mut fields_to_be_added, ctx)
+        calculate_fields_for_prisma_join_table(table, &mut fields_to_be_added, &datamodel)
     }
 
     for (model, field) in fields_to_be_added {
-        ctx.datamodel
-            .find_model_mut(&model)
-            .add_field(Field::RelationField(field));
+        datamodel.find_model_mut(&model).add_field(Field::RelationField(field));
+    }
+
+    if !sanitization_leads_to_duplicate_names(&datamodel) {
+        // our opinionation about valid names
+        sanitize_datamodel_names(ctx, &mut datamodel);
+    }
+
+    // deduplicating relation field names
+    deduplicate_relation_field_names(&mut datamodel);
+
+    if !ctx.previous_datamodel.is_empty() {
+        enrich(ctx.previous_datamodel, &mut datamodel, ctx, warnings);
+        debug!("Enriching datamodel is done.");
     }
 
-    Ok(())
+    // commenting out models, fields, enums, enum values
+    warnings.append(&mut commenting_out_guardrails(&mut datamodel, ctx));
+
+    // try to identify whether the schema was created by a previous Prisma version
+    let version = version_checker::check_prisma_version(ctx, warnings);
```

Comment:
This gives false info if we do not check some of the warnings before calling it, that's why adding it here and not where we call the introspection. Maybe a worth of a revisit soonish?